replacer.go

```diff
@@ -354,7 +354,15 @@ func (f fileReplacementProvider) replace(key string) (any, bool) {
 			zap.Error(err))
 		return nil, true
 	}
-	return string(body), true
+	content := string(body)
+	// Remove a single trailing newline, regardless of its type
+	if strings.HasSuffix(content, "\r\n") {
+		return strings.TrimSuffix(content, "\r\n"), true
+	}
+	if strings.HasSuffix(content, "\n") {
+		return strings.TrimSuffix(content, "\n"), true
+	}
+	return strings.TrimSuffix(content, "\r"), true
```

Comment:
```suggestion
	content := string(body)
	content = strings.TrimSuffix(content, "\n")
	content = strings.TrimSuffix(content, "\r")
	return content, true
```

Was suggested already.

Reply:
If we're being pedantic, you can use `bytes` package instead of the `strings` package, then only convert it into string on return :)

Reply:
Maybe, I've never wrote a go code before just for a context 🤣